roachtest: acceptance/version-upgrade failure

In [this CI run](https://teamcity.cockroachdb.com/viewLog.html?buildId=2525418&buildTypeId=Cockroach_UnitTests), I saw the following seg fault:
```
panic: runtime error: invalid memory address or nil pointer dereference
[signal SIGSEGV: segmentation violation code=0x1 addr=0x0 pc=0x12a2516]

goroutine 40 [running]:
panic(0x1e35b60, 0x519d690)
	/usr/local/go/src/runtime/panic.go:1064 +0x545 fp=0xc000deb758 sp=0xc000deb690 pc=0x4504a5
runtime.panicmem(...)
	/usr/local/go/src/runtime/panic.go:212
runtime.sigpanic()
	/usr/local/go/src/runtime/signal_unix.go:742 +0x413 fp=0xc000deb788 sp=0xc000deb758 pc=0x467073
github.com/cockroachdb/cockroach/pkg/sql/types.(*T).Family(...)
	/go/src/github.com/cockroachdb/cockroach/pkg/sql/types/types.go:1124
github.com/cockroachdb/cockroach/pkg/sql/types.(*T).Equivalent(0xc0002109c0, 0x0, 0x0)
	/go/src/github.com/cockroachdb/cockroach/pkg/sql/types/types.go:1752 +0x36 fp=0xc000deb7d0 sp=0xc000deb788 pc=0x12a2516
github.com/cockroachdb/cockroach/pkg/workload/schemachange.(*operationGenerator).setColumnDefault(0xc000eede90, 0xc000dae750, 0x18, 0xc0003730a8, 0xc, 0x0)
	/go/src/github.com/cockroachdb/cockroach/pkg/workload/schemachange/operation_generator.go:1395 +0x2ab fp=0xc000deb960 sp=0xc000deb7d0 pc=0x187878b
github.com/cockroachdb/cockroach/pkg/workload/schemachange.(*operationGenerator).randOp(0xc000eede90, 0xc000dae750, 0x1, 0x5, 0x0, 0x7fbcf46fc1f8, 0x20, 0x28)
	/go/src/github.com/cockroachdb/cockroach/pkg/workload/schemachange/operation_generator.go:230 +0x20b fp=0xc000deba78 sp=0xc000deb960 pc=0x186e52b
github.com/cockroachdb/cockroach/pkg/workload/schemachange.(*schemaChangeWorker).runInTxn(0xc000eedef0, 0xc000dae750, 0x0, 0x0, 0xc000dae750, 0x0)
	/go/src/github.com/cockroachdb/cockroach/pkg/workload/schemachange/schemachange.go:239 +0xff fp=0xc000debc38 sp=0xc000deba78 pc=0x18826ff
github.com/cockroachdb/cockroach/pkg/workload/schemachange.(*schemaChangeWorker).run(0xc000eedef0, 0x3fc7c80, 0xc000eeeec0, 0x0, 0x0)
	/go/src/github.com/cockroachdb/cockroach/pkg/workload/schemachange/schemachange.go:322 +0x125 fp=0xc000debea8 sp=0xc000debc38 pc=0x1883d05
github.com/cockroachdb/cockroach/pkg/workload/schemachange.(*schemaChangeWorker).run-fm(0x3fc7c80, 0xc000eeeec0, 0x0, 0x0)
	/go/src/github.com/cockroachdb/cockroach/pkg/workload/schemachange/schemachange.go:313 +0x3e fp=0xc000debee0 sp=0xc000debea8 pc=0x18871de
github.com/cockroachdb/cockroach/pkg/workload/cli.workerRun(0x3fc7c80, 0xc000eeeec0, 0xc0002f6c00, 0xc0003c37c0, 0x0, 0xc0007a31d0)
	/go/src/github.com/cockroachdb/cockroach/pkg/workload/cli/run.go:237 +0xaf fp=0xc000debf58 sp=0xc000debee0 pc=0x13f432f
github.com/cockroachdb/cockroach/pkg/workload/cli.runRun.func2.1(0xc0006f6000, 0xc0003dc080, 0xc0002f6c00, 0x0, 0x3fc7c80, 0xc000eeeec0, 0xc0003c37c0, 0x0, 0xc0007a31d0)
	/go/src/github.com/cockroachdb/cockroach/pkg/workload/cli/run.go:426 +0xec fp=0xc000debf98 sp=0xc000debf58 pc=0x13f82cc
```
This occurred when running the `schemachange` workload:
```
run_041420.658_n1_workload_run_schemachange: 04:14:20 cluster.go:386: > /go/src/github.com/cockroachdb/cockroach/bin/roachprod run local:1 -- ./workload run schemachange --verbose=1 --tolerate-errors=true --max-ops 10 --concurrency 2 {pgurl:1-4}
```
cc @jayshrivastava, since I think you're the one working most closely in this area these days.

--- a/pkg/sql/rowenc/testutils.go
+++ b/pkg/sql/rowenc/testutils.go
@@ -1086,7 +1086,7 @@ func RandCreateTables(
 		if i > 0 && rng.Intn(2) == 0 {
 			interleave = tables[rng.Intn(i)].(*tree.CreateTable)
 		}
-		t := RandCreateTableWithInterleave(rng, prefix, i+1, interleave)
+		t := RandCreateTableWithInterleave(rng, prefix, i+1, interleave, nil)
 		tables[i] = t
 	}
 
@@ -1099,13 +1099,25 @@ func RandCreateTables(
 
 // RandCreateTable creates a random CreateTable definition.
 func RandCreateTable(rng *rand.Rand, prefix string, tableIdx int) *tree.CreateTable {
-	return RandCreateTableWithInterleave(rng, prefix, tableIdx, nil)
+	return RandCreateTableWithInterleave(rng, prefix, tableIdx, nil, nil)
+}
+
+// RandCreateTableWithColumnIndexNumberGenerator creates a random CreateTable definition
+// using the passed function to generate column index numbers for column names.
+func RandCreateTableWithColumnIndexNumberGenerator(
+	rng *rand.Rand, prefix string, tableIdx int, generateColumnIndexNumber func() int64,
+) *tree.CreateTable {
+	return RandCreateTableWithInterleave(rng, prefix, tableIdx, nil, generateColumnIndexNumber)
 }
 
 // RandCreateTableWithInterleave creates a random CreateTable definition,
 // interleaved into the given other CreateTable definition.
 func RandCreateTableWithInterleave(
-	rng *rand.Rand, prefix string, tableIdx int, interleaveInto *tree.CreateTable,
+	rng *rand.Rand,
+	prefix string,
+	tableIdx int,
+	interleaveInto *tree.CreateTable,
+	generateColumnIndexNumber func() int64,
 ) *tree.CreateTable {
 	// columnDefs contains the list of Columns we'll add to our table.
 	nColumns := randutil.RandIntInRange(rng, 1, 20)
@@ -1148,7 +1160,11 @@ func RandCreateTableWithInterleave(
 			// Loop until we generate an indexable column type.
 			var extraCol *tree.ColumnTableDef
 			for {
-				extraCol = randColumnTableDef(rng, tableIdx, i+prefixLength)
+				colIdx := i + prefixLength
+				if generateColumnIndexNumber != nil {
+					colIdx = int(generateColumnIndexNumber())
+				}
+				extraCol = randColumnTableDef(rng, tableIdx, colIdx)
 				extraColType := tree.MustBeStaticallyKnownType(extraCol.Type)
 				if colinfo.ColumnTypeIsIndexable(extraColType) {
 					break
@@ -1185,7 +1201,11 @@ func RandCreateTableWithInterleave(
 	} else {
 		// Make new defs from scratch.
 		for i := 0; i < nColumns; i++ {
-			columnDef := randColumnTableDef(rng, tableIdx, i)
+			colIdx := i
+			if generateColumnIndexNumber != nil {
+				colIdx = int(generateColumnIndexNumber())
+			}
+			columnDef := randColumnTableDef(rng, tableIdx, colIdx)
 			columnDefs = append(columnDefs, columnDef)
 			defs = append(defs, columnDef)
 		}

--- a/pkg/workload/schemachange/operation_generator.go
+++ b/pkg/workload/schemachange/operation_generator.go
@@ -526,7 +526,7 @@ func (og *operationGenerator) createTable(tx *pgx.Tx) (string, error) {
 		return "", err
 	}
 
-	stmt := rowenc.RandCreateTable(og.params.rng, "table", tableIdx)
+	stmt := rowenc.RandCreateTableWithColumnIndexNumberGenerator(og.params.rng, "table", tableIdx, og.newUniqueSeqNum)
 	stmt.Table = *tableName
 	stmt.IfNotExists = og.randIntn(2) == 0
 

